base/setup/lib/fsutil.c

```diff
@@ -0,0 +1,418 @@
+//
+// This is basically the code for listing available FileSystem providers
+// (currently hardcoded in a list), and for performing a basic FileSystem
+// recognition for a given disk partition.
+//
+// See also: https://git.reactos.org/?p=reactos.git;a=blob;f=reactos/dll/win32/fmifs/init.c;h=e895f5ef9cae4806123f6bbdd3dfed37ec1c8d33;hb=b9db9a4e377a2055f635b2fb69fef4e1750d219c
+// for how to get FS providers in a dynamic way. In the (near) future we may
+// consider merging some of this code with us into a fmifs / fsutil / fslib library...
+//
+
+/* INCLUDES *****************************************************************/
+
+#include "precomp.h"
+
+#include "fsutil.h"
+#include "partlist.h"
+
+#include <fslib/vfatlib.h>
+// #include <fslib/ext2lib.h>
+// #include <fslib/ntfslib.h>
+
+#define NDEBUG
+#include <debug.h>
+
+
+FILE_SYSTEM RegisteredFileSystems[] =
+{
+    { L"FAT"  , VfatFormat, VfatChkdsk },
+//  { L"FAT32", VfatFormat, VfatChkdsk },
+#if 0
```

Comment:
Technically it would be great if the FAT formatter allowed for specifying the FAT format (FAT 16, 32) to use (but it doesn't atm). I've instead added an explicit comment about this fact in the code.